otel/propagator.go

```diff
@@ -45,10 +45,12 @@ func (p sentryPropagator) Inject(ctx context.Context, carrier propagation.TextMa
 
 	// Propagate baggage header
 	sentryBaggageStr := ""
-	if sentrySpan != nil && sentrySpan.IsTransaction() {
-		// TODO(anton): Normally, this should return the DSC baggage from the transaction
-		// (not necessarily the current span). So this might break things in some cases.
-		sentryBaggageStr = sentrySpan.ToBaggage()
+	if sentrySpan != nil {
+		// Dynamic sampling context is set only on the (root) transaction.
+		sentryTransaction := sentrySpan.GetTransaction()
+		if sentryTransaction != nil {
+			sentryBaggageStr = sentryTransaction.ToBaggage()
+		}
```

Comment:
This is the actual fix.

Reply:
Tbh, I would actually fix `ToBaggage()`.

```go

// ToBaggage returns the serialized dynamic sampling context in the baggage format.
func (s *Span) ToBaggage() string {
	if s.IsTransaction() {
		return s.GetTransaction().dynamicSamplingContext.String()
	}

	return s.dynamicSamplingContext.String()
}
```